test/wasi/test-wasi.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+require('../common');
+
+const cp = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+
+const SKIP = [
+  'poll',
+];
+
+const C_DIR = path.resolve(__dirname, 'c');
+const WASM_DIR = path.resolve(__dirname, 'wasm');
+
+function parseOptions(string) {
+  const options = {};
+  for (const line of string.split('\n')) {
+    if (!line.startsWith('//')) {
+      break;
+    }
+    const [key, value] = line.slice(3).split(': ');
+    if (!key || !value) {
+      continue;
+    }
+    options[key] = value.replace(/\\n/g, '\n');
+  }
+  return options;
+}
+
+function exec(command, { stdin } = {}) {
+  try {
+    const stdout = cp.execSync(command, {
+      input: stdin,
+    });
+    return { code: 0, stdout: stdout.toString() };
+  } catch (e) {
+    if (e.status) {
+      return { code: e.status, stdout: '' };
+    }
+    throw e;
+  }
+}
+
+const ARGS = [
+  '--expose-internals',
+  `-r ${__dirname}/patch_preopen`,
+  '--experimental-wasm-bigint',
```

Comment:
Currently experimental-wasm-bigint doesn't work on `arm, android_arm, android_ia32, ia32, ppc, s390, s390x, mipsel, mips`.

Reply:
hmm yeah that would be a bit of a blocker.

Reply:
That blocker can be removed by not using the i64 -> BigInt conversion. Also would it be possible to commit Wast/Wat instead and using wabt (or smiliar) to convert it to Wasm?

Reply:
wasi uses i64, so it's out of my hands.

> would it be possible to commit Wast/Wat

Any particular reason? Using wasi-sysroot gives us better coverage because many system calls are actually multiple wasi calls, plus it's good to test against real world binaries.

Reply:
My experience is that maintaining binaries is not easy; requires disassemble them to get the content.

Reply:
well I uploaded the c sources, so that shouldn't be a problem

Reply:
My preference would also be toward using wasi-sysroot as long as none of the tests require re-compiling.

Reply:
> well I uploaded the c sources, so that shouldn't be a problem
 
Sorry forgot about that :+1: